### ISSUE TITLE
Serializing Date failed new with 2.6.1

### ISSUE THREAD

**Issue by jans23** [FasterXML/jackson-module-scala#222]
The following test fails with jackson-module-scala 2.6.1 but succeeded in earlier versions including 2.6.0-1:

```scala
private val dateFormat = new java.text.SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'")
val myDate = DateClass(dateFormat.parse("2013-03-02T23:56:27Z"))
val myDateJson = "{\n  \"my_date\" : \"2013-03-02T23:56:27Z\"\n}"

test("Serialize Date object to JSON string") {
  assertResult(myDateJson) { serialize(myDate) }
}
```

Test results:

> Serialize Date object to JSON string *** FAILED ***
> Expected "...ate" : "2013-03-02T2[3]:56:27Z"
> }", but got "...ate" : "2013-03-02T2[2]:56:27Z"

I use the following wrapper:

```scala
object JacksonWrapper extends LazyLogging {
  private val dateFormat = new java.text.SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'")
  val mapper = new ObjectMapper() with ScalaObjectMapper
  mapper.registerModule(DefaultScalaModule)
  mapper.setDateFormat(dateFormat)

  // ...
}
```

**Comment by cowtowncoder** [comment]
Is this specific to Scala module? Or does it also fail with standard databind `ObjectMapper`?
If latter, it should be filed under `jackson-databind`.

**Comment by jans23** [comment]
I don't know because I don't use Java.

**Comment by christophercurrie** [comment]
@cowtowncoder are there changes in databind that might be related here? IIRC there aren't any changes in the Scala module between 2.6.0-1 and 2.6.1.

**Comment by cowtowncoder** [comment]
@christophercurrie The only change that seems even remotely related would be:

https://github.com/FasterXML/jackson-databind/issues/889

but I don't see how it would directly affect things here.

**Comment by christophercurrie** [comment]
The fact that it's an hour off is super suspicious. DST wasn't in effect on 3/2/2013, so that's probably not involved, but a single hour of difference reeks of time zone wonkiness. I haven't had a chance to try and reproduce the result locally, but it might be helpful to know what the value of the `user.timezone` system property is when you're running the test.

**Comment by christophercurrie** [comment]
Didn't check the cross reference until just now. It looks like this is fixed in `jackson-databind` 2.6.3; we'll get a matching Scala module release ASAP, but in the interim the versions should be cross-compatible.

**Comment by jans23** [comment]
I can confirm that this issue is fixed in 2.6.3. Thank you.
